Sources/Mock.swift

```diff
@@ -101,14 +104,15 @@ public struct Mock: Equatable {
     /// The callback which will be executed everytime this `Mock` was started. Can be used within unit tests for validating that a request has been started. The callback must be set before calling `register`.
     public var onRequest: OnRequest?
     
-    private init(url: URL? = nil, ignoreQuery: Bool = false, dataType: DataType, statusCode: Int, data: [HTTPMethod: Data], additionalHeaders: [String: String] = [:], fileExtensions: [String]? = nil) {
+    private init(url: URL? = nil, ignoreQuery: Bool = false, reportFailure: Bool = false, dataType: DataType, statusCode: Int, data: [HTTPMethod: Data], additionalHeaders: [String: String] = [:], fileExtensions: [String]? = nil) {
```

Comment:
Wouldn't it make more sense to rename this like:

```suggestion
    private init(url: URL? = nil, ignoreQuery: Bool = false, requestError: Error? = nil, dataType: DataType, statusCode: Int, data: [HTTPMethod: Data], additionalHeaders: [String: String] = [:], fileExtensions: [String]? = nil) {
```

This allows to set explicit types of errors and allows more usecases. We could then simply verify whether this value is set and report a failure instead if so.